.github/workflows/build_test_release.yaml

```diff
@@ -0,0 +1,91 @@
+name: Reusable build, test and release
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        required: true
+        type: string
+      runner:
+        required: true
+        type: string
+      compat-tests:
+        default: false
+        type: boolean
+      release-artefact:
+        default: ''
+        type: string
+    secrets:
+      token:
+        required: true
+
+jobs:
+  build-test:
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      runner: ${{ inputs.runner }}
+      upload-artifact: ${{ inputs.release-artefact }}
+      repository: pytorch/multipy
+      script: |
+        # Build
+        xargs sudo dnf install -y <build-requirements-centos8.txt
+        pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
+        pip install -e .
+        python multipy/runtime/example/generate_examples.py
+        
+        # Test
```

Comment:
Is it possible to split these up into different jobs? You should be able to create a dependency from the build and the other jobs. Just make sure tarball creation + distribution is dependent on the tests passing.

Reply:
Yes, that would be possible! However, each step using the `linux-job` from `pytest/test-infra` begins with a plain [conda-builder](https://hub.docker.com/r/pytorch/conda-builder) image. Therefore, each would need to install dependencies and build MultiPy separately which is quite expensive I'm afraid.

I added what I believe to be the next best thing which is log groups (sections that can be collapsed):

<img width="722" alt="image" src="https://user-images.githubusercontent.com/19345882/207746616-7572033e-9025-40c0-8de4-40c13fb78bfd.png">

Reply:
ohh that's great. We just want to make sure we have a good way to split things up :)